### DISCUSSION TITLE
Using CFW-NG Bootloader to switch between (old) CFW and OFW?

### DISCUSSION THREAD

**Discussion by Nebukadneza** [toniebox-reverse-engineering/hackiebox_cfw_ng#33] [Q&A] [answered]
Hi There,

thanks for your great work on this cute little box — currently I’m trying to understand the state of these projects to evaluate whether I want to buy a box.

The main thing I’m wondering about: The (old, non-NG) CFW had a web-interface that allowed upload to SD, right? However, the new NG CFW currently only consists of a (much improved) bootloader, if I see the repos state correctly? Now, is my assumption correct that i could implement a workflow like this:
- Flash NG Bootloader
- Use 1 Slot for OFW, 1 Slot for the old non-NG CFW
- Boot OFW by default
- Boot CFW (using the ears-pressing-dance) only when I want to use the web-interface for file-upload

Is this a use-case currently supported? To elaborate on what I want to do: I’m only interested in creating custom figures with my own tags (additionally to buying official ones), to play libre/open/private content that is not officially available, and does not infringe any copyright. For this, it’d be great if I didn’t have to open up the box for SD-access for all changes, hence my question regarding the web-interface.

Thank you for your work! ^_^

**Comment by SciLor** [comment, marked as answer]
Hey Nebukadneza,
the CFW has a Webinterface with an upload function. You can use it to upload big files, but that isn't recommend, because this will be very slow for technical reasons.

Usually we install a microSD extension in the Toniebox to overcome this problem to have easy access to the microSD. It will be installed in the bottom of the box so you'll just need to open the bottom cap.

With the CFW NG Bootloader you were alright, but for your usecase you don't need it. I really suggest you to visit our [Telegram group](https://t.me/toniebox_reverse_engineering) for future questions and a faster response.

Bee
